### ISSUE TITLE
Big Sur window border doesn't stay on top of window

### ISSUE THREAD

**Issue by peterjbachman** [koekeishiya/yabai#713]
I updated to Big Sur last night and I can't seem to figure out why the window border goes behind the window after clicking on the window

Picture for reference. The blue is the focused window, but I haven't clicked on it yet. You can see the corners of the window border on the terminal tile.
<img width="1680" alt="Screen Shot 2020-11-13 at 9 13 26 AM" src="https://user-images.githubusercontent.com/46987067/99094267-7ff0ac00-25cb-11eb-9e0b-2fc555f18620.png">

I tried uninstalling, restarting and installing yabai again and the issue doesn't seem to be resolved.

Using `tail -f /usr/local/var/log/yabai/yabai.err.log` there is an error stating `unknown command 'insert_window_border_color' for domain 'config'`

Currently on OS 11.0.1 and Yabai 3.3.3

**Comment by 24unix** [comment]
There ist no option insert_window_border_color.

`╰─➤  grep window_border yabai-3.3.1/src/message.c                                                                            2 ↵
#define COMMAND_CONFIG_BORDER                "window_border"
#define COMMAND_CONFIG_BORDER_WIDTH          "window_border_width"
#define COMMAND_CONFIG_BORDER_ACTIVE_COLOR   "active_window_border_color"
#define COMMAND_CONFIG_BORDER_NORMAL_COLOR   "normal_window_border_color"
            fprintf(rsp, "%s\n", bool_str[g_window_manager.enable_window_border]);
            window_manager_set_window_border_enabled(&g_window_manager, false);
            window_manager_set_window_border_enabled(&g_window_manager, true);
                window_manager_set_window_border_width(&g_window_manager, width + ((width&0x1) ? 1 : 0));
                window_manager_set_active_window_border_color(&g_window_manager, color);
                window_manager_set_normal_window_border_color(&g_window_manager, color);
            window_manager_toggle_window_border(&g_window_manager, acting_window);
`

**Comment by peterjbachman** [comment]
> There is no option insert_window_border_color.
> 
> `╰─➤ grep window_border yabai-3.3.1/src/message.c 2 ↵ #define COMMAND_CONFIG_BORDER "window_border" #define COMMAND_CONFIG_BORDER_WIDTH "window_border_width" #define COMMAND_CONFIG_BORDER_ACTIVE_COLOR "active_window_border_color" #define COMMAND_CONFIG_BORDER_NORMAL_COLOR "normal_window_border_color" fprintf(rsp, "%s\n", bool_str[g_window_manager.enable_window_border]); window_manager_set_window_border_enabled(&g_window_manager, false); window_manager_set_window_border_enabled(&g_window_manager, true); window_manager_set_window_border_width(&g_window_manager, width + ((width&0x1) ? 1 : 0)); window_manager_set_active_window_border_color(&g_window_manager, color); window_manager_set_normal_window_border_color(&g_window_manager, color); window_manager_toggle_window_border(&g_window_manager, acting_window); `

That appeared to be something old from my .yabairc config file.

When I run the commands 
`sudo yabai --uninstall-sa
sudo yabai --install-sa
sudo yabai --load-sa`

I do get the error `could not retrieve task port for pid: XXXXX
yabai: scripting-addition failed to inject payload into Dock.app!`

It appears that after I reinstalled Yabai, it isn't asking for permission to access Dock.app. I don't know how to give Yabai access to Dock.app, and it doesn't appear as an option on the "Automation" page in my Security Preferences.

**Comment by 24unix** [comment]
I also had problems today after upgrade to Buf Sur and Yabai 3.3.1.
Did you codesign (scripts/codesign) /path/to/yabai the binary?

**Comment by peterjbachman** [comment]
Yeah I did codesign the binary. It still fails to inject the payload into Dock.app.

I mean if it's anything I am getting a notification that it fails to inject the payload now, and I wasn't getting that before. 🤷‍♂️

**Comment by 24unix** [comment]
Please perform all those steps separately. Which of them throws the error? The install or the load part?

**Comment by peterjbachman** [comment]
Okay so I completely uninstalled Yabai, wiped my Accessibility and automation privileges and restarted my computer. Then I did the following:

`brew install koekeishiya/formulae/yabai`
No error
`sudo yabai --install-sa`
No error
I also gave Yabai Accessibility Privileges at this point
`brew services start yabai`
No error, but I'm still encountering the border issue.
`sudo yabai --load-sa`
This is where I get the error:
`could not retrieve task port for pid: XXXXXX
yabai: scripting-addition failed to inject payload into Dock.app!`

I also followed the Big Sur instructions and added `sudo yabai --load-sa` into my config file while also allowing that to run without a password.

At no point was I prompted to allow Yabai access to Dock.app

**Comment by koekeishiya** [comment]
Dis you disable SIP?

**Comment by peterjbachman** [comment]
> Did you disable SIP?

I ran `csrutil --without debug --without fs` as per the installation instructions.
Here's my `csrutil status`:

`Configuration:`
`Apple Internal: disabled`
`Kext Signing: enabled`
`Filesystem Protections: disabled`
`Debugging Restrictions: enabled`
`DTrace Restrictions: enabled`
`NVRAM Protections: enabled`
`BaseSystem Verification: enabled`

I'll disable SIP completely and see if that changes anything.

**Comment by peterjbachman** [comment]
Everything appears to work after completely disabling SIP. There are still some issues with the border here and there, but restarting the Yabai service seems to fix those issues.

**Comment by jclusso** [comment]
@peterjbachman why was this closed when the border issue still exists. Even if you restart the border issue exists essentially works only the first window focused.

**Comment by peterjbachman** [comment]
@jclusso Here's the steps I took that fixed this issue:

1. Disable SIP in recovery mode using `csrutil disable --with kext --with dtrace --with nvram --with basesystem`
2. Follow [https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(latest-release)](url) to bypass sudo privileges necessary to run `sudo yabai --load-sa` when Yabai is loaded, and add the command to your .yabairc

I have not run into any issues with the border since I did this.

**Comment by jclusso** [comment]
@peterjbachman I have it entirely disabled already. Maybe it has to do with the fact that I'm running Apple Silicon?
